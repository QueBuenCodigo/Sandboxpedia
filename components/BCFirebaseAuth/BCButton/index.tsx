import React from "react";
import styled from "styled-components"

interface Props {
  variant?: string;
  className?:string;
  onClick?:(event: React.MouseEvent<HTMLButtonElement>)=>void;
}

const BCButton: React.FC<Props> = (props) => {
  const { variant, className, onClick } = props;
  return <Button onClick={onClick} className={className}>{props.children}</Button>;
};

const Button = styled.button`
    background:none;
    border:0;
    cursor:pointer;
    outline:0;
`;

export default BCButton;
