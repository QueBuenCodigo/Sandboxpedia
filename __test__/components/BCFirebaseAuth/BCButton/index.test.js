import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import BCButton from '../../../../components/BCFirebaseAuth/BCButton/index';

describe('<BCButton/>',()=>{

    const button=mount(<BCButton/>)

    test('test del renderizado de BCButton',()=>{

        expect(button.length).toEqual(1);
    });
});


describe('BCButton Snapshot',()=>{

    test('comprobar la UI del componente BCButton',()=>{
        const button= create(<BCButton/>);
        expect(button.toJSON()).toMatchSnapshot();
    });

});

