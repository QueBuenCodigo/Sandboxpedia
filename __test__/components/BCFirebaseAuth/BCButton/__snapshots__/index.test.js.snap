// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BCButton Snapshot comprobar la UI del componente BCButton 1`] = `
<button
  className="BCButton__Button-sc-18tvwvx-0 lgdhbd"
/>
`;
